taskvine/src/manager/vine_current_transfers.c

```diff
@@ -52,6 +54,19 @@ char *vine_current_transfers_add(struct vine_manager *q, struct vine_worker_info
 int vine_current_transfers_remove(struct vine_manager *q, const char *id)
 {
 	struct vine_transfer_pair *p;
+	p = hash_table_lookup(q->current_transfer_table, id);
+
+	// remove the pending replica from the worker's file table
+	if (p && p->to && p->source_url) {
+		const char *cachename = path_basename(p->source_url);
```

Comment:
This looks error prone. Could we keep a pointer to the corresponding replica when creating the vine_transfer_pair?